eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -368,21 +377,26 @@ public boolean equals(Object obj) {
 
         private String namespace;
 
-        private Builder(InstanceInfo result, VipAddressResolver vipAddressResolver) {
+        private Builder(InstanceInfo result, VipAddressResolver vipAddressResolver, Function<String,String> intern) {
             this.vipAddressResolver = vipAddressResolver;
             this.result = result;
+            this.intern = intern != null ? intern : StringCache::intern;
```

Comment:
All three use cases are still internal to this class, so it seems reasonable to explicitly specify at each place. The benefit we'd get is that the if check would no longer need to be performed at each builder creation.